encoding/rs/icicle.go

```diff
@@ -0,0 +1,35 @@
+//go:build icicle
+
+package rs
+
+import (
+	"github.com/Layr-Labs/eigenda/encoding"
+	"github.com/Layr-Labs/eigenda/encoding/fft"
+	"github.com/Layr-Labs/eigenda/encoding/icicle"
+	rs_icicle "github.com/Layr-Labs/eigenda/encoding/rs/icicle"
+)
+
+const (
+	defaultNTTSize = 25 // Used for NTT setup in Icicle backend
+)
+
+func CreateIcicleBackendEncoder(e *Encoder, params encoding.EncodingParams, fs *fft.FFTSettings) (*ParametrizedEncoder, error) {
+	icicleDevice, err := icicle.NewIcicleDevice(icicle.IcicleDeviceConfig{
+		GPUEnable: e.Config.GPUEnable,
```

Comment:
It looks Icicle will support also CPU now?

Reply:
Yes but it's not as performant as the gnark backend, am trying to look into why